### DISCUSSION TITLE
getAddresses

### DISCUSSION THREAD

**Discussion by NetFreak26** [wevm/viem#1568] [Question] [answered]
I am using viem@1.2.12. I created a wallet client like this:

```
    const account = privateKeyToAccount(privateKey as Hex);
    const walletClient = createWalletClient({
        account: account,
        chain: chain,
        transport: http(rpcUrl)
    }); 
```

But when I am trying to fetch the address with walletClient.getAddresses(), it is returning an empty array.
I have check the account exist with the private key I am providing.

**Comment by jxom** [comment, marked as answer]
`getAddresses` calls `eth_accounts` to the RPC provider internally. You are probably after `account.address`.
